### PR TITLE
Add recipe for `dataeval`

### PR DESCRIPTION
**Pull request by aria-ml**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by aria-ml** [comment]
@conda-forge/help-python ready for review! 😄

File: recipes/dataeval/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+{% set version = "0.61.0" %}
+{% set sha256 = "bb79f19913a3e6f4732cbba08f9d71006c11cf917081b6b81649a65ee5c39e4f" %}
+
+package:
+  name: "{{ name|lower }}"
+  version: "{{ version }}"
+
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
```
Comment (ocefpaf) [inline recipes/dataeval/meta.yaml:11]:
```suggestion

package:
  name: {{ name|lower }}
  version: {{ version }}

source:
  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
  sha256: bb79f19913a3e6f4732cbba08f9d71006c11cf917081b6b81649a65ee5c39e4f
```

File: recipes/dataeval/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  run:
+    - python >=3.9, <3.12
+    - cython <3
+    - hdbscan
+    - maite
+    - numpy
+    - pillow
+    - python-xxhash
+    - scipy
+    - scikit-learn
```
Comment (ocefpaf) [inline recipes/dataeval/meta.yaml:34]: These are the dependencies according the package metadata. Maybe those are wrong but, if that is so, we need to open an issue upstream.

```suggestion
    - python >=3.9,<3.12
    - poetry-core >=1.0.0
    - poetry-dynamic-versioning
    - pip
  run:
    - python >=3.9,<3.12
    - hdbscan >=0.8.36
    - numpy >1.24.3
    - pillow >=10.3.0
    - scipy >=1.10
    - scikit-learn >=1.5.0
    - python-xxhash >=3.3
    - maite
  run_constrained:
    - matplotlib-base *
    - nvidia-cudnn-cu11 >=8.6.0.163
    - tensorflow >=2.14.1,<2.16
    - tensorflow-io-gcs-filesystem >=0.35.0,<0.37
    - tensorflow_probability >=0.22.1,<0.24
    - pytorch >=2.0.1,!=2.2.0
```

**Review by ocefpaf — COMMENTED** [review]
Please use grayskull to generate Python recipes. Grayskull make it easier for us to review.

See https://github.com/conda-forge/staged-recipes?tab=readme-ov-file#grayskull---recipe-generator-for-python-packages-on-pypi for more information.

File: recipes/dataeval/meta.yaml (hunk 1 above)
Comment (aria-ml) [reply recipes/dataeval/meta.yaml]: I'm glad grayskull works now!  When we had a more complicated dependency list (due to python 3.8 support) grayskull would raise exceptions trying to resolve the dependency tree.

**Pull request merged**